web/main.ts

```diff
@@ -2520,6 +2565,7 @@ class GitGraphView {
 			(codeReviewPossible ? '<div id="cdvCodeReview" class="cdvControlBtn">' + SVG_ICONS.review + '</div>' : '') +
 			(!expandedCommit.loading ? '<div id="cdvFileViewTypeTree" class="cdvControlBtn cdvFileViewTypeBtn" title="File Tree View">' + SVG_ICONS.fileTree + '</div><div id="cdvFileViewTypeList" class="cdvControlBtn cdvFileViewTypeBtn" title="File List View">' + SVG_ICONS.fileList + '</div>' : '') +
 			(externalDiffPossible ? '<div id="cdvExternalDiff" class="cdvControlBtn">' + SVG_ICONS.linkExternal + '</div>' : '') +
+			(loadedParents.length > 1 ? '<div id="cdvParentToggle" class="cdvControlBtn" title="Toggle Parent">' + SVG_ICONS.merge + '</div>' : '') +

```

Comment:
Instead of this button being a toggle that cycles through the parents, it might be clearer if it opened a context menu that allows the user to select a specific parent.

For example, the context menu options could be something like: `<parent-index> <abbrev-parent-hash>: <subject-if-available>`

Reply:
This is what I implemented first, but the context menu wouldn't work. I was about to write you quite a lengthy message when I figured out that if I use left click to open a menu it would open and then immediately close as left click also triggered the closing code. This made me switch to the current toggle implementation.

Since then, I've ran into `Event.stopPropagation` which may be the correct solution here, but I would like to hear your opinion.

Reply:
Yes, I think you'll need `Event.stopPropagation` to use a context menu on left click.

Reply:
Just to clarify, what do you mean by "subject if available"?

> For example, the context menu options could be something like: `<parent-index> <abbrev-parent-hash>: <subject-if-available>`

Do you mean something like `commitDetails.body.split("\n")[0]`, and make sure it's not empty so there won't be a colon at the end?

Reply:
I was referring to subject of the parent commit, so the user can read what each parent is (instead of looking at the parent index & commit hash). You can use the `this.commits` & `this.commitLookup` to retrieve the parent commit’s subject (but only if the commit is loaded in the view).

Hopefully this is clearer: `<parent-index> <abbrev-parent-hash>: <parent-subject-if-parent-is-loaded> `